### PR TITLE
add birthdate validation in Pet Controller

### PR DESCRIPTION
**Pull request by 4313testing**
adding birthdate for update and create Pet

### CODE REVIEW

File: src/main/java/org/springframework/samples/petclinic/owner/PetController.java [hunk 1]
```diff
@@ -87,6 +88,10 @@ public String processCreationForm(Owner owner, @Valid Pet pet, BindingResult res
 		if (StringUtils.hasLength(pet.getName()) && pet.isNew() && owner.getPet(pet.getName(), true) != null) {
 			result.rejectValue("name", "duplicate", "already exists");
 		}
+		LocalDate currentDate = LocalDate.now();
+		if (pet.getBirthDate() != null && pet.getBirthDate().isAfter(currentDate)) {
+			result.rejectValue("birthDate", "invalid", "Invalid birth date. Please select a valid date.");
```
Comment (dsyer) [inline src/main/java/org/springframework/samples/petclinic/owner/PetController.java:93]: The error code should probably be more unique than simply "invalid"? Maybe "invalid.date" (and then add a message to `messages.properties`).

Comment (4313testing) [reply src/main/java/org/springframework/samples/petclinic/owner/PetController.java:93]: okay now its done you can check it in the new commit

File: src/main/resources/messages/messages_de.properties [hunk 2]
```diff
@@ -6,3 +6,4 @@ nonNumeric=darf nur numerisch sein
 duplicateFormSubmission=Wiederholtes Absenden des Formulars ist nicht erlaubt
 typeMismatch.date=ung�ltiges Datum
 typeMismatch.birthDate=ung�ltiges Datum
+invalid.date=Ongeldige geboortedatum. Selecteer alstublieft een geldige datum.
```
Comment (dsyer) [inline src/main/resources/messages/messages_de.properties:9]: This isn't German.

Comment (4313testing) [reply src/main/resources/messages/messages_de.properties]: german message is fixed now !

File: src/main/java/org/springframework/samples/petclinic/owner/PetController.java [hunk 3]
```diff
@@ -107,6 +112,18 @@ public String initUpdateForm(Owner owner, @PathVariable("petId") int petId, Mode
 
 	@PostMapping("/pets/{petId}/edit")
 	public String processUpdateForm(@Valid Pet pet, BindingResult result, Owner owner, ModelMap model) {
+
+		if (StringUtils.hasLength(pet.getName()) && pet.isNew() && owner.getPet(pet.getName(), true) != null) {
```
Comment (dsyer) [inline src/main/java/org/springframework/samples/petclinic/owner/PetController.java:116]: Is this necessary - it's not a new pet, so the test will always fail.

Comment (4313testing) [reply src/main/java/org/springframework/samples/petclinic/owner/PetController.java]: you are correct this was not there in the latest version .
sorry i maybe i added it by mistake 
its fixed now in the new commit

**Comment by dsyer** [comment]
Can you please squash everything down to one commit and rebase on main?

**Comment by dsyer** [comment]
Merged as https://github.com/spring-projects/spring-petclinic/commit/3be289517d320a47bb8f359acc1d1daf0829ed0b

**Pull request closed without merging**